### PR TITLE
M2 features

### PR DESCRIPTION
**Pull request by abenso**
Parser and UI Implementation
- Parameters
- Memo
- Spend
- Output
- Swap
- ICS-20 Withdrawal

Effect hash computation
Spend authorization signature
Updated icons

### CODE REVIEW

**Review by chcmedeiros — APPROVED** [review]

**Comment by emmanuelm41** [comment]
On crypto.c, I see we are caching the keys, but I don't see where we unset the cache
```
if (!fvk_cached_set) {
        // Compute seed
        CATCH_ZX_ERROR(computeSpendKey(&keys));

        // use seed to compute viewieng keys
        CATCH_ZX_ERROR(compute_keys(&keys));

        // Copy keys
        CATCH_ZX_ERROR(copyKeys(&keys, Fvk, output, len, cmdResponseLen));

        MEMCPY(fvk_cached, keys.fvk, FVK_LEN);

        fvk_cached_set = true;
    } else {
        MEMCPY(output, fvk_cached, FVK_LEN);
    }
```

**Comment by emmanuelm41** [comment]
On coin.h, the ticket is "penumbra". I think it should be UM 

```
#define COIN_TICKER "penumbra"
```

**Comment by emmanuelm41** [comment]
On ICS20 transfers, are we holding a list of supported tokens, or anything like that? I wonder how we handling amount formatting, and tickers

**Comment by emmanuelm41** [comment]
On zemu tests, for show address randomized, on flex/stax we see "sub account # 1" when the device allows you to display it as a QR code... later you see the address. Is that correct? It seems a bit weird

**Comment by emmanuelm41** [comment]
For unknown assets, what we usually do is to display the asset id in a different page, not in the same where the fee value is displayed. It is a bit hard to read the fee alongside with the token asset id (taken from zemu images to)

**Comment by neithanmo** [comment]
> On zemu tests, for show address randomized, on flex/stax we see "sub account # 1" when the device allows you to display it as a QR code... later you see the address. Is that correct? It seems a bit weird


Yes, that was a request from them, to display sub-account #address_index

**Comment by neithanmo** [comment]
> For unknown assets, what we usually do is to display the asset id in a different page, not in the same where the fee value is displayed. It is a bit hard to read the fee alongside with the token asset id (taken from zemu images to)


I think the same, but that is the specific UI they requested in the test vectors
like everything in the same line.

**Comment by emmanuelm41** [comment]
From past experience, I think we will need to leave the unknown asset ids under expert mode, as ledger won't accept those long asset ids as clear signing. If expert mode is not enabled, we should reject the tx with a specific error

**Review by emmanuelm41 — CHANGES_REQUESTED** [review]
left some comments

File: app/Makefile.version [hunk 1]
```diff
@@ -3,4 +3,4 @@ APPVERSION_M=0
 # This is the minor version
 APPVERSION_N=0
 # This is the patch version
-APPVERSION_P=2
```
[the remark this answers is not in the document]
Comment (neithanmo) [reply app/Makefile.version:6]: indeed, i will bump it to 0.1.0

**Review by emmanuelm41 — APPROVED** [review]

**Pull request merged**